### PR TITLE
When the window is summoned and is already active, minimize it.

### PR DESCRIPTION
**Pull request by zadjii-msft**
This adds a `toggleVisibility` parameter to `globalSummon`. 
* When `true` (default): when you press the global summon keybinding, and the window is currently the foreground window, we'll minimize the window.
* When `false`, we'll just do nothing.

## References
* Original thread: #653
* Spec: #9274 
* megathread: #8888

## PR Checklist
* [x] Checks a box in #8888
* [x] closes https://github.com/microsoft/terminal/projects/5#card-59030814
* [x] I work here
* [ ] No tests for this one.
* [ ] yes yes eventually I'll come back on the docs

## Detailed Description of the Pull Request / Additional comments

I've got nothing extra to add here. This one's pretty simple. I'm only targeting #9954 since that one laid so much foundation to build on, with the `SummonBehavior`

## Validation Steps Performed

Played with this for a while, and it's amazing.

### CODE REVIEW

**Review by carlos-zamora — APPROVED** [review]

**Review by DHowett — APPROVED** [review]

**Comment by zadjii-msft** [comment]
> tfw this pr has 2 signoffs but #9954 before it has none, and #9854 _before that one_ only has one

**Comment by DHowett** [comment]
> > tfw this pr has 2 signoffs but #9954 before it has none, and #9854 _before that one_ only has one

small atomic PRs are nice and easy :wink:

**Comment by DHowett** [comment]
I'm fixing the x86 leg break

File: src/cascadia/UnitTests_Remoting/RemotingTests.cpp [hunk 1]
```diff
@@ -38,17 +38,20 @@ namespace RemotingUnitTests
...
         }
-        HRESULT MoveWindowToDesktop(HWND /*topLevelWindow*/, REFGUID /*desktopId*/)
+        IFACEMETHOD(MoveWindowToDesktop)
+        (HWND /*topLevelWindow*/, REFGUID /*desktopId*/)
         {
             VERIFY_IS_TRUE(false, L"We shouldn't need GetWindowDesktopId in the tests.");
             return E_FAIL;
         }
-        HRESULT IsWindowOnCurrentVirtualDesktop(HWND topLevelWindow, BOOL* onCurrentDesktop)
+        IFACEMETHOD(IsWindowOnCurrentVirtualDesktop)
```
Comment (DHowett) [inline src/cascadia/UnitTests_Remoting/RemotingTests.cpp:53]: yes clang-format did this no i don't like it but oh well

**Pull request merged**